src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/NewBlogPostStoringActor.java

```diff
@@ -1,46 +1,57 @@
 import pl.tomaszdziurko.jvm_bloggers.utils.DateTimeUtilities;
 
-import java.time.LocalDateTime;
 import java.util.Date;
-import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
 
 @Slf4j
+@ExtensionMethod({DateTimeUtilities.class, StringUtils.class})
 public class NewBlogPostStoringActor extends AbstractActor {
 
-    private final BlogPostRepository blogPostRepository;
-
     public NewBlogPostStoringActor(BlogPostRepository blogPostRepository) {
-        this.blogPostRepository = blogPostRepository;
-
         receive(ReceiveBuilder.match(RssEntryWithAuthor.class, rssEntry -> {
-                Optional<BlogPost> existingPost = blogPostRepository.findByUrl(rssEntry.getRssEntry().getLink());
-                if (!existingPost.isPresent()) {
-                    storeNewBlogPost(rssEntry);
-                } else {
-                    log.trace("Existing post found, skipping save");
-                }
-            }
-        ).build());
+                BlogPost blogPost = blogPostRepository
+                    .findByUrl(rssEntry.getRssEntry().getLink())
+                    .orElseGet(() -> createBlogPost(rssEntry));
+                updateDescription(blogPost, rssEntry.getRssEntry().getDescription());
+                blogPostRepository.save(blogPost);
+            }).build()
+        );
     }
 
-    private void storeNewBlogPost(RssEntryWithAuthor rssEntry) {
+    private BlogPost createBlogPost(RssEntryWithAuthor rssEntry) {
         SyndEntry postInRss = rssEntry.getRssEntry();
-        Date dateToStore = postInRss.getPublishedDate() != null ? postInRss.getPublishedDate() : postInRss.getUpdatedDate();
-        LocalDateTime publishedDate = DateTimeUtilities.convertDateToLocalDateTime(dateToStore);
-        BlogPost newBlogPost = new BlogPost(postInRss.getTitle(), rssEntry.getBlog(), postInRss.getLink(), publishedDate,
-            rssEntry.getBlog().getDefaultApprovedValue());
-        blogPostRepository.save(newBlogPost);
-        log.info("Stored new post '{}' with id {} by {}", newBlogPost.getTitle(), newBlogPost.getId(), rssEntry.getBlog().getAuthor());
+        Date dateToStore = firstNonNull(postInRss.getPublishedDate(), postInRss.getUpdatedDate());
+        log.info("Creating new post '{}' by {}", postInRss.getTitle(), rssEntry.getBlog().getAuthor());
+        return BlogPost.builder()
+                .title(postInRss.getTitle())
+                .url(postInRss.getLink())
+                .publishedDate(dateToStore.convertDateToLocalDateTime())
+                .approved(rssEntry.getBlog().getDefaultApprovedValue())
+                .blog(rssEntry.getBlog())
+                .build();
+    }
+
+    private void updateDescription(BlogPost blogPost, SyndContent descriptionContent) {
+        if (descriptionContent != null) {
+            final String description = descriptionContent.getValue().abbreviate(BlogPost.MAX_DESCRIPTION_LENGTH);
```

Comment:
Trimming description to fit [`BlogPost#description`](https://github.com/tdziurko/jvm-bloggers/pull/111/files#diff-cc7a92da5546ac7e5a7bf75d639eedbfR47) field.